db/seeds.rb

```diff
@@ -11,3 +11,31 @@
0,"title_x":50,"title_y":13,"titleEnabled":true},"allNodes":[{"x":100,"y":20,"type":1,"bitWidth":1,"label":"","connections":[13]},{"x":100,"y":40,"type":1,"bitWidth":1,"label":"","connections":[11]},{"x":0,"y":20,"type":0,"bitWidth":1,"label":"","connections":[4]},{"x":0,"y":40,"type":0,"bitWidth":1,"label":"","connections":[5]},{"x":10,"y":0,"type":1,"bitWidth":1,"label":"","connections":[2]},{"x":10,"y":0,"type":1,"bitWidth":1,"label":"","connections":[3]},{"x":10,"y":0,"type":1,"bitWidth":1,"label":"","connections":[10]},{"x":100,"y":20,"type":1,"bitWidth":1,"label":"","connections":[12]},{"x":100,"y":40,"type":1,"bitWidth":1,"label":"","connections":[16]},{"x":0,"y":20,"type":0,"bitWidth":1,"label":"","connections":[11]},{"x":0,"y":40,"type":0,"bitWidth":1,"label":"","connections":[6]},{"x":510,"y":250,"type":2,"bitWidth":1,"label":"","connections":[1,9]},{"x":650,"y":290,"type":2,"bitWidth":1,"label":"","connections":[7,14]},{"x":-20,"y":-10,"type":0,"bitWidth":1,"label":"","connections":[0]},{"x":-20,"y":10,"type":0,"bitWidth":1,"label":"","connections":[12]},{"x":20,"y":0,"type":1,"bitWidth":1,"label":"","connections":[17]},{"x":10,"y":0,"type":0,"bitWidth":1,"label":"","connections":[8]},{"x":10,"y":0,"type":0,"bitWidth":1,"label":"","connections":[15]}],"id":17468728326,"name":"Full Adder","Input":[{"x":340,"y":230,"objectType":"Input","label":"","direction":"RIGHT","labelDirection":"LEFT","propagationDelay":0,"customData":{"nodes":{"output1":4},"values":{"state":1},"constructorParamaters":["RIGHT",1,{"x":0,"y":20,"id":"eMapSKxw4lGTR9wemDsp"}]}},{"x":340,"y":250,"objectType":"Input","label":"","direction":"RIGHT","labelDirection":"LEFT","propagationDelay":0,"customData":{"nodes":{"output1":5},"values":{"state":1},"constructorParamaters":["RIGHT",1,{"x":0,"y":40,"id":"837tujwUlLFHp8nlQgTN"}]}},{"x":340,"y":310,"objectType":"Input","label":"","direction":"RIGHT","labelDirection":"LEFT","propagationDelay":0,"customData":{"nodes":{"output1":6},"values":{"state":0},"constructorParamaters":["RIGHT",1,{"x":0,"y":60,"id":"PuthdFyas7NQ66FMFOqd"}]}}],"Output":[{"x":670,"y":310,"objectType":"Output","label":"","direction":"LEFT","labelDirection":"RIGHT","propagationDelay":0,"customData":{"nodes":{"inp1":16},"constructorParamaters":["LEFT",1,{"x":100,"y":20,"id":"Af0TIhW3aLK88scEqtqf"}]}},{"x":720,"y":240,"objectType":"Output","label":"","direction":"LEFT","labelDirection":"RIGHT","propagationDelay":0,"customData":{"nodes":{"inp1":17},"constructorParamaters":["LEFT",1,{"x":100,"y":40,"id":"fVe0eexwXF7A8QKIC60t"}]}}],"XorGate":[{"x":670,"y":240,"objectType":"XorGate","label":"","direction":"RIGHT","labelDirection":"LEFT","propagationDelay":100,"customData":{"constructorParamaters":["RIGHT",2,1],"nodes":{"inp":[13,14],"output1":15}}}],"SubCircuit":[{"x":390,"y":210,"id":"38912139787","inputNodes":[2,3],"outputNodes":[0,1],"version":"2.0"},{"x":510,"y":270,"id":"38912139787","inputNodes":[9,10],"outputNodes":[7,8],"version":"2.0"}],"nodes":[11,12]}]}',
```

Comment:
I don't think it is a good idea to have the json data inline with the code. Can you have this in a json file fullAdder.json and read it during runtime? @tachyons might have some inputs

Reply:
ok

Reply:
Yeah, loading from a json file will be much better